Add TodoListItem component and implement useState for newTodo display

henrymauricio1987 opened the pull request:
Weekly Assignment – TodoListItem Component & useState

✅ Created a TodoListItem.jsx component to render individual todos.

✅ Updated TodoList.jsx to map todos using the new TodoListItem component with proper key props.

✅ Implemented useState in App.jsx to manage a newTodo string value.

✅ Rendered the newTodo state inside a paragraph between the TodoForm and TodoList.

This completes the expected functionality for this week’s assignment. ✔️

JoHeyJo commented on src/TodoList.jsx:
```diff
@@ -0,0 +1,27 @@
+    ];
+
+    return (
+        <div>
```
Why are we mapping and rendering `todos` twice?

henrymauricio1987 replied:
Sorry about that! I just wanted to test what would show up, but it ended up looking the same. I'll go ahead and remove it—thanks!

JoHeyJo commented on src/TodoListItem.jsx:
```diff
@@ -0,0 +1,6 @@
+function TodoListItem({ todo }) {
+    return <li>{todo.id}{todo.space}{todo.title} </li>
```
Is `todo.space` the best way to achieve creating a space inside and HTML element? Remember that you can render strings inside of HTML element along side expressions.

henrymauricio1987 commented on src/App.jsx:
```diff
@@ -1,35 +1,27 @@
+        <TodoList />
+
+        <p>{newTodo}</p>
+        <p>{setNewTodo}</p>
```
I was just testing to see what would render, but nothing showed up. I'll remove it now. Thanks!

henrymauricio1987 commented:
Ah, got it—sorry about that! I'm a hands-on learner, so I duplicated the todo map to better understand how the rendering works and added that space property to make things clearer visually. But you're totally right—I've already cleaned up both issues based on your feedback. I'll keep DevTools open to catch any sneaky bugs moving forward. Really appreciate the help!
